### PR TITLE
Add runner with arg parsing

### PR DESCRIPTION
**Pull request by k88hudson-cfa**
I still need to add custom arg parsing, but here is a start to a runner with some base arguments and generic setup functionality.

### CODE REVIEW

File: src/runner.rs [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+use crate::global_properties::ContextGlobalPropertiesExt;
+use crate::random::ContextRandomExt;
+use crate::report::ContextReportExt;
+use clap::{Args, Command, FromArgMatches as _};
+
+#[derive(Args, Debug)]
+pub struct BaseArgs {
+    /// Random seed
+    #[arg(short, long, default_value = "0")]
+    pub seed: u64,
```
Comment (ekr-cfa) [inline src/runner.rs:14]: I would probably call this `random_seed`

File: src/runner.rs [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    /// Optional path for report output
+    #[arg(short, long, default_value = "")]
+    pub output_dir: String,
+}
+
+#[derive(Args)]
+pub struct PlaceholderCustom {}
+
+#[allow(clippy::missing_errors_doc)]
+pub fn run_with_custom_args<A, F>(load: F) -> Result<(), Box<dyn std::error::Error>>
```
Comment (ekr-cfa) [inline src/runner.rs:34]: I don't love "load" as a name. Maybe "setup_fn"?

File: src/runner.rs [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+where
+    A: Args,
+    F: Fn(&mut Context, BaseArgs, Option<A>) -> Result<(), IxaError>,
+{
+    let cli = Command::new("Ixa");
```
Comment (ekr-cfa) [inline src/runner.rs]: Shouldn't this instead be "ixa"

File: src/runner.rs [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    let base_args_matches = BaseArgs::from_arg_matches(&matches)?;
+    let custom_matches = A::from_arg_matches(&matches)?;
+
+    run_with_args_internal(base_args_matches, Some(custom_matches), load)
+}
+
+#[allow(clippy::missing_errors_doc)]
+pub fn run_with_args<F>(load: F) -> Result<(), Box<dyn std::error::Error>>
+where
+    F: Fn(&mut Context, BaseArgs, Option<PlaceholderCustom>) -> Result<(), IxaError>,
```
Comment (ekr-cfa) [inline src/runner.rs:51]: Can you just do Option<()>. It doesnt' seem like there are any trait bounds.

File: src/runner.rs [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+    args: BaseArgs,
+    custom_args: Option<A>,
+    load: F,
+) -> Result<(), Box<dyn std::error::Error>>
+where
+    A: Args,
+    F: Fn(&mut Context, BaseArgs, Option<A>) -> Result<(), IxaError>,
+{
+    // Create a context
+    let mut context = setup_context(&args)?;
```
Comment (ekr-cfa) [inline src/runner.rs:92]: Is there a reason not to inline setup_context, which seems to only be called from ehre.

File: src/runner.rs [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+    struct CustomArgs {
+        #[arg(short, long, default_value = "0")]
+        field: u32,
+    }
+
+    #[test]
+    fn test_run_with_custom_args() {
+        let result = run_with_custom_args(|_, _, _: Option<CustomArgs>| Ok(()));
+        assert!(result.is_ok());
+    }
```
Comment (ekr-cfa) [inline src/runner.rs:118]: These tests don't seem to verify that any of the arguments exist or are correct. It seems like that's really challenging, though.

**Review by ekr-cfa — CHANGES_REQUESTED** [review]

File: src/runner.rs (hunk 4 above)
Comment (k88hudson-cfa) [reply src/runner.rs:51]: I have `A` implementing `Args` but i guess for this function it's not actually necessary

File: src/runner.rs (hunk 6 above)
Comment (k88hudson-cfa) [reply src/runner.rs:111]: it would probably be good to have some integration tests yes, would you be ok if I file that

**Review by ekr-cfa — APPROVED** [review]
LGTM

**Pull request merged**